apps/cbportal/base/cloudbroker__cloudspace/methodclass/cloudbroker_cloudspace.py

```diff
@@ -136,6 +136,42 @@ def removeIP(self, cloudspaceId, ipaddress, **kwargs):
         """
         return True
 
+    @auth(['level1', 'level2', 'level3'])
+    @wrap_remote
+    def getVFW(self, cloudspaceId, **kwargs):
+        """
+        Get VFW info
+        param:cloudspaceId id of the cloudspace
+        """
+        cloudspaceId = int(cloudspaceId)
+        if not self.models.cloudspace.exists(cloudspaceId):
+            raise exceptions.NotFound('Cloudspace with id %s not found' % (cloudspaceId))
+
+        cloudspace = self.models.cloudspace.get(cloudspaceId)
+        fwid = '%s_%s' % (cloudspace.gid, cloudspace.networkId)
+
+        network = self.vfwcl.virtualfirewall.get(fwid)
```

Comment:
you might wonna do an exists check here aswell as it wont work for CS in virtual or destroyed state